sources/host-ctr/cmd/host-ctr/main.go

```diff
@@ -879,8 +905,17 @@ func tagImage(ctx context.Context, imageName string, newImageName string, client
 }
 
 // withDynamicResolver provides an initialized resolver for use with ref.
-func withDynamicResolver(ctx context.Context, ref string) containerd.RemoteOpt {
-	noOp := func(_ *containerd.Client, _ *containerd.RemoteContext) error { return nil }
+func withDynamicResolver(ctx context.Context, ref string, registryConfig *RegistryConfig) containerd.RemoteOpt {
```

Comment:
The implementation here has different behavior for private ECR registries compared to all others.  I don't really _expect_ folks to want to use mirrors for ECR, but we should either call that out, update the [resolver](https://github.com/awslabs/amazon-ecr-containerd-resolver/) we're using to support it, or revisit the choice of using that resolver...

Reply:
I'll go ahead an add a comment here to explain that private ECR registries will not use the registry configuration.

I can create an issue in amazon-ecr-containerd-resolver's repository for registry mirror support. I wasn't sure if there is an use-case for it.